docs/Routes.md

```diff
@@ -210,18 +232,14 @@ fastify.get('/', options, async function (request, reply) {
 <a name="promise-resolution"></a>
 ### Promise resolution
 
-If your handler is an `async` function or returns a promise, you should be aware of a special behaviour which is necessary to support the callback and promise control-flow. If the handler's promise is resolved with `undefined`, it will be ignored causing the request to hang and an *error* log to be emitted.
+If your handler is an `async` function or returns a promise, you should be aware of a special behaviour which is necessary to support the callback and promise control-flow. If the handler's promise is resolved with `undefined`, it will be ignored causing the request to hang and an *error* log to be emitted. Follow these rules and you are safe:
 
-1. If you want to use `async/await` or promises but respond a value with `reply.send`:
-    - **Don't** `return` any value.
-    - **Don't** forget to call `reply.send`.
-2. If you want to use `async/await` or promises:
-    - **Don't** use `reply.send`.
-    - **Don't** return `undefined`.
+1. If you want to use `async/await` **Don't** use `reply.send` and just return the value so that the `route` handler function is resolved with the value you want to respond.
+2. If you want to use `reply.send` **Don't** use `async/await` and **Don't** return a promise otherwise the control-flow will get lost and the `route` handler function is resolved before you can call `reply.send`.
```

Comment:
We actually support a 3rd way to mix and match, i.e. `await reply`. We might want to mention as follows: "If you have a complex use case where you need to mix them, just add `await reply` at the end of your async function".